src/dir_entry.rs

```diff
@@ -82,10 +82,12 @@ impl DirEntry {
             .as_ref()
     }
 
-    pub fn depth(&self) -> Option<usize> {
+    /// Returns true if the entry has the depth greater or equal compared to `depth`
+    pub fn deeper(&self, depth: usize) -> bool {
         match &self.inner {
-            DirEntryInner::Normal(e) => Some(e.depth()),
-            DirEntryInner::BrokenSymlink(_) => None,
+            DirEntryInner::Normal(e) => e.depth() >= depth,
+            // broken symlink passes match if it is just longer thatn the `depth`
+            DirEntryInner::BrokenSymlink(p) => p.components().count() - 1 >= depth,
```

Comment:
What if the starting directory has multiple complements?

Reply:
Yes, the result depends on the size of the root component too so my approach wouldn't work.

It will erroneously match the `link`
    
    fd -L link foo/bar/baz --min-depth 3

while the symmetric call for non-link would exclude match:

    mkdir -p foo/bar/baz/fourth
    fd -L fourth foo/bar/baz --min-depth 3

`ignore` crate hides too much information so it's problematic to extract root.